pilot/pkg/config/coredatamodel/controller.go

```diff
@@ -128,9 +133,7 @@ func (c *Controller) Apply(change *sink.Change) error {
 		return fmt.Errorf("descriptor type not supported %s", change.Collection)
 	}
 
-	c.syncedMu.Lock()
-	c.synced[change.Collection] = true
-	c.syncedMu.Unlock()
+	c.sync(change.Collection)
```

Comment:
At what point is it correct to consider the collection "synced"? Is it as soon as we receive the event, or after we've processed it and updated our structures (e..g c.configStore)?

Reply:
I believe this sync is more from MCP perspective to indicate that pilot know about the resource type. For a full sync (envoy has received it?) I believe there is a status endpoint which could be queried.